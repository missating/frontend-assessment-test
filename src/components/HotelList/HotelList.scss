.hotellist-container {
  width: 90%;
  margin: 0 auto;
  max-width: 900px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 2rem;

  &__content {
    width: 100%;
    display: flex;
    margin: 20px;

    &-input {
      width: 100%;
      height: 45px;
      box-sizing: border-box;
      outline: none;
      border: 1px solid lightgray;
      border-radius: 3px;
      padding: 0.5rem;
      font-size: 0.5rem;
    }

      &-select-box {
      position: relative;
      max-width: 400px;

      select {
        border: 1px solid lightgray;
        outline:none;
        background: transparent;
        -webkit-appearance: none;
        -moz-appearance: none;
        appearance: none;
        border-radius: 0;
        width: 100%;
        cursor: pointer;
        padding: 12px 55px 15px 15px;
        font-size: 0.5rem;
        height: 45px;
      }

      &:after {
        position: absolute;
        right:0; top:0;
        width: 40px;
        height: 45px;
        line-height: 50px;
        content: '\25BC';
        text-align: center;
        z-index: -1;
        background: #000000;;
        color: #ffffff;
        font-size: 0.5rem;
      }
    }
  }
}
