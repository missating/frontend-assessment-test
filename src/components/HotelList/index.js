// third party libraries
import React, { useEffect, useState } from 'react';

// components
import HotelCard from '../HotelCard'

// styles
import './HotelList.scss'

const API_URL = 'http://fake-hotel-api.herokuapp.com/api/hotels'

const HotelList = () => {
  const [hotels, setHotels] = useState([]);

  useEffect(() => {
    fetch(`${API_URL}?count=10`)
      .then((response) => response.json())
      .then((response) => {
        setHotels(response)
        console.log(response)
      })
      .catch((error) => {
        throw error
      })
  }, [])


  return (
    <div className="hotellist-container">
      {
        hotels && hotels.map((data) => {
          const extractStartDate = new Date(data.date_start).toDateString()
          const extractEndDate = new Date(data.date_end).toDateString()
          return (
            <HotelCard
              hotelId={data.id}
              key={data.id}
              image={data.images[0]}
              name={data.name}
              description={data.description}
              price={data.price}
              star={data.stars}
              city={data.city}
              country={data.country}
              start={extractStartDate}
              end={extractEndDate}
            />
          )
        })
      }
    </div>
  )
}

export default HotelList;
