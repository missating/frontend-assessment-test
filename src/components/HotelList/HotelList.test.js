// react library
import React from 'react';
import { shallow } from 'enzyme';

// components
import HotelList from './index'

describe('HotelList Component', () => {
  it('should be rendered properly', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const wrapper = shallow(<HotelList />);
    expect(wrapper.find('div').length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
  });
})
