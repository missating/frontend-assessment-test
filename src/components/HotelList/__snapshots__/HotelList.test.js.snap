// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList Component should be rendered properly 1`] = `
<div
  className="hotellist-container"
>
  <div
    className="hotellist-container__content"
  >
    <select
      className="hotellist-container__content-select"
      name="select-box"
      onChange={[Function]}
      value={3}
    >
      <option>
        1
      </option>
      <option>
        2
      </option>
      <option>
        3
      </option>
      <option>
        4
      </option>
      <option>
        5
      </option>
    </select>
    <input
      className="hotellist-container__content-input"
      onChange={[Function]}
      placeholder="Filter By Price"
      type="number"
    />
  </div>
</div>
`;
